lib/metadata/include/irods/metadata.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef IRODS_METADATA_HPP
+#define IRODS_METADATA_HPP
+
+#include <string>
+#include <vector>
+#include <map>
```

Comment:
Move these after the `#ifdef`-blocks.

The convention for iRODS is:
- iRODS includes
- iRODS externals includes
- System includes
- C++ std library includes (with a preference for C includes first)